internal/daemon/plugin/vendorplugin.go

```diff
@@ -106,9 +109,11 @@ func WithVspImage(template_vars map[string]string) func(*GrpcPlugin) {
 		vspImage := template_vars["VendorSpecificPluginImage"]
 		d.vspImage = vspImage
 		d.log.Info("VSP Image", "vspImage", d.vspImage)
-		err := render.ApplyAllFromBinData(d.log, "vsp-ds", template_vars, binData, d.k8sClient, nil, nil)
-		if err != nil {
-			d.log.Error(err, "Failed to start vendor plugin container", "vspImage", d.vspImage)
+		if vspImage != "" {
```

Comment:
After discussion with Sal.
This is quite confusing especially when the function name is called `WithVspImage`
We should call this code in a new function called `DeployVspImage` which is called when we create a new Grpc Plugin.

Reply:
this `With*` function is a "Functional Options Pattern" in golang.

Me personally, I am dubious about whether this is really a great pattern, but it's at least a wide spread and common pattern.

In this case, the functions that return the callbacks are commonly called `WithSomething()`. `DeployVspImage` seems thus a bad name.

Feel free to open a patch to whatever you wish.

Reply:
PTAL at @168

Reply:
PTAL at @https://github.com/openshift/dpu-operator/pull/168